### PR TITLE
PM-1812 Add a Cart Line Item to Existing Single Requisition List

### PR DESCRIPTION
**Pull request by petro97**
Description:

Linting:

- [x] No linting errors

Component Updates:

- [ ] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:

- [x] E2E tests (npm test run with `e2e`)
- [x] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: src/components/src/CartMain/cart.main.tsx [hunk 1]
```diff
@@ -41,63 +45,361 @@ interface CartMainProps {
...
+  }
+
+  handleAddToList() {
+    const { itemData, listArray } = this.state;
+    const { onShowAlert } = this.props;
+    this.setState({ addToRequisitionListLoading: true });
+    if (listArray.length) {
+      const body: { [key: string]: any } = {};
+      body.items = itemData;
+      const promices = listArray.map(listUrl => cortexFetch(listUrl, {
```
Comment (shaunmaharaj) [inline src/components/src/CartMain/cart.main.tsx]: typo: promices

File: src/components/src/CartMain/cart.main.tsx [hunk 2]
```diff
@@ -41,63 +45,361 @@ interface CartMainProps {
...
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify({
+          name: listName,
+        }),
+      })
+        .then((res) => {
+          if (res.status === 200 || res.status === 201) {
```
Comment (shaunmaharaj) [inline src/components/src/CartMain/cart.main.tsx]: Will an error still display as a banner if this fails?

Comment (petro97) [reply src/components/src/CartMain/cart.main.tsx]: @shaunmaharaj I have updated this. If this fails, there will display message container with error and a modal window will close

**Comment by shaunmaharaj** [comment]
Great work with this one @petro97 @yulia-dnistrian!!! 👏 👏 👏 
Merging now! 😃

**Pull request merged**